wpical/src/main/native/cpp/fieldmap.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include "fieldmap.h"
+
+#include <Eigen/Dense>
+#include <Eigen/Geometry>
+
+fieldmap::fieldmap() {}
+
+fieldmap::fieldmap(wpi::json json) {
```

Comment:
```suggestion
fieldmap::fieldmap(const wpi::json& json) {
```

This should take the json by const ref.  Changing this does mean the code below changes, though, and you'll have to use the potentially-throwing const getters and catch those.